packages/@aws-cdk/aws-ivs-alpha/lib/recording-configuration.ts

```diff
@@ -0,0 +1,366 @@
+  }
+
+  private _renderRenditionConfiguration(): CfnRecordingConfiguration.RenditionConfigurationProperty | undefined {
+    if (!this.props.renditions && !this.props.renditionSelection) {
+      return;
+    }
+
+    return {
+      renditions: this.props.renditions,
+      renditionSelection: this.props.renditionSelection,
+    };
+  };
+
+  private _renderThumbnailConfiguration(): CfnRecordingConfiguration.ThumbnailConfigurationProperty | undefined {
+    if (!this.props.thumbnailRecordingMode
+      && !this.props.thumbnailResolution
+      && !this.props.thumbnailStorage
+      && !this.props.thumbnailTargetInterval) {
+      return;
+    }
+
+    return {
+      recordingMode: this.props.thumbnailRecordingMode,
+      resolution: this.props.thumbnailResolution,
+      storage: this.props.thumbnailStorage,
+      targetIntervalSeconds: this.props.thumbnailTargetInterval?.toSeconds(),
+    };
+  };
+
+  private validateRecordingConfigurationName(): undefined {
+    const recordingConfigurationName = this.props.recordingConfigurationName;
+
+    if (recordingConfigurationName == undefined || Token.isUnresolved(recordingConfigurationName)) {
+      return;
+    }
+
+    if (!/^[a-zA-Z0-9-_]*$/.test(recordingConfigurationName)) {
+      throw new Error(`\`recordingConfigurationName\` must consist only of alphanumeric characters, hyphens or underbars, got: ${recordingConfigurationName}.`);
+    }
+
+    if (recordingConfigurationName.length > 128) {
+      throw new Error(`\`recordingConfigurationName\` must be less than or equal to 128 characters, got: ${recordingConfigurationName.length} characters.`);
+    }
+  };
+
+  private validateRecordingReconnectWindowSeconds(): undefined {
+    const recordingReconnectWindow = this.props.recordingReconnectWindow;
+
+    if (recordingReconnectWindow === undefined || Token.isUnresolved(recordingReconnectWindow)) {
+      return;
+    }
+
+    if (0 < recordingReconnectWindow.toMilliseconds() && recordingReconnectWindow.toMilliseconds() < Duration.seconds(1).toMilliseconds()) {
+      throw new Error(`\`recordingReconnectWindow\` must be between 0 and 300 seconds, got ${recordingReconnectWindow.toMilliseconds()} milliseconds.`);
+    }
+
+    if (recordingReconnectWindow.toSeconds() > 300) {
+      throw new Error(`\`recordingReconnectWindow\` must be between 0 and 300 seconds, got ${recordingReconnectWindow.toSeconds()} seconds.`);
+    }
+  };
+
+  private validateRenditionSettings(): undefined {
+    if (this.props.renditionSelection === RenditionSelection.CUSTOM && !this.props.renditions) {
+      throw new Error('`renditions` must be provided when \`renditionSelection\` is `RenditionSelection.CUSTOM`.');
+    }
+
+    if (this.props.renditionSelection !== RenditionSelection.CUSTOM && this.props.renditions) {
+      throw new Error(`\`renditions\` can only be set when \`renditionSelection\` is \`RenditionSelection.CUSTOM\`, got ${this.props.renditionSelection}.`);
+    }
```

Comment:
This validation can be removed if we use an enum-like class instead of the enum for `rendtitionSelection` and enforce when `renditions` needs to be provided in the parameters of which selection is chosen.

Reply:
Removed validations.